ngraph/test/backend/matrix_nms.in.cpp

```diff
@@ -0,0 +1,930 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// clang-format off
+#ifdef ${BACKEND_NAME}_FLOAT_TOLERANCE_BITS
+#define DEFAULT_FLOAT_TOLERANCE_BITS ${BACKEND_NAME}_FLOAT_TOLERANCE_BITS
+#endif
+
+#ifdef ${BACKEND_NAME}_DOUBLE_TOLERANCE_BITS
+#define DEFAULT_DOUBLE_TOLERANCE_BITS ${BACKEND_NAME}_DOUBLE_TOLERANCE_BITS
+#endif
+// clang-format on
+
+#include "gtest/gtest.h"
+#include "runtime/backend.hpp"
+#include "ngraph/runtime/tensor.hpp"
+#include "ngraph/ngraph.hpp"
+#include "util/all_close.hpp"
+#include "util/all_close_f.hpp"
+#include "util/known_element_types.hpp"
+#include "util/ndarray.hpp"
+#include "util/test_control.hpp"
+#include "util/test_tools.hpp"
+
+NGRAPH_SUPPRESS_DEPRECATED_START
+
+using namespace std;
+using namespace ngraph;
+
+static string s_manifest = "${MANIFEST}";
+
+NGRAPH_TEST(${BACKEND_NAME}, matrix_nms_output_type_i64)
+{
+    std::vector<float> boxes_data = {0.0, 0.0,  1.0, 1.0,  0.0, 0.1,   1.0, 1.1,
+                                     0.0, -0.1, 1.0, 0.9,  0.0, 10.0,  1.0, 11.0,
+                                     0.0, 10.1, 1.0, 11.1, 0.0, 100.0, 1.0, 101.0};
+
+    std::vector<float> scores_data = {
+        0.9, 0.75, 0.6, 0.95, 0.5, 0.3,
+        0.95, 0.75, 0.6, 0.80, 0.5, 0.3};
+
+    op::v8::MatrixNms::Attributes attrs;
+    attrs.nms_top_k = 3;
+    attrs.score_threshold = 0.0f;
+    attrs.sort_result_type = op::v8::MatrixNms::SortResultType::SCORE;
+    attrs.keep_top_k = -1;
+    attrs.background_class = 0;
+
+    const auto boxes_shape = Shape{1, 6, 4};  // N 1, C 2, M 6
+    const auto scores_shape = Shape{1, 2, 6};
+
+    const auto boxes = make_shared<op::Parameter>(element::f32, boxes_shape);
+    const auto scores = make_shared<op::Parameter>(element::f32, scores_shape);
+    attrs.decay_function = op::v8::MatrixNms::DecayFunction::LINEAR;
+    attrs.gaussian_sigma = 2.0f;
+    attrs.post_threshold = 0.0f;
+
+    auto nms = make_shared<op::v8::MatrixNms>(boxes, scores, attrs);
+
+    auto f = make_shared<Function>(nms, ParameterVector{boxes, scores});
+
+    auto backend = runtime::Backend::create("${BACKEND_NAME}");
```

Comment:
We are making efforts to refactor backend tests which use legacy Backend to use [TestCase](https://github.com/openvinotoolkit/openvino/blob/master/ngraph/test/util/test_case.hpp) class instead.
Please refactor backend tests to use TestCase. Thead [here](https://github.com/openvinotoolkit/openvino/pull/6292#pullrequestreview-689269109).
cc @tomdol

Reply:
Done.